### PR TITLE
Breaking: Update hint to check for XCTO header on all resources

### LINKED ISSUE #1221: Require `X-Content-Type-Options` headers for all responses? [3]
Following your recommendation (through the webapp), we kept the "X-Content-Type-Options" response header only for JavaScript and CSS resources. However, another recommendation of the community (in this case Scott Helme) is to use this same header for all content types, though MDN seems to agree with you rather that Scott.

Example here:
https://securityheaders.com/?q=https%3A%2F%2Fwww.instantluxe.com%2F&followRedirects=on

I'm confused as to how this header should be used.

### PR DESCRIPTION
**Pull request by utsavized**
This reverts changes added to check for this header only
on scripts and stylesheets, and instead, checks for the header
on all resources. MDN suggests the former, but Chromium uses
this response header on more than script/stylesheets for
CORB.

- - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - -

Ref https://github.com/webhintio/hint/issues/1221

Close #1221

## Pull request checklist

Make sure you:

- [x] Signed the [Contributor License Agreement](https://cla.js.foundation/webhintio/hint)
- [x] Followed the [commit message guidelines](https://webhint.io/docs/contributor-guide/getting-started/pull-requests/#commit-messages)

For non-trivial changes, please make sure you also:

- [x] Added/Updated related documentation.
- [x] Added/Updated related tests.

### CODE REVIEW

**Comment by utsavized** [comment]
@molant Added documentation of server config. Thanks!

File: packages/hint-x-content-type-options/README.md [hunk 1]
```diff
@@ -126,11 +190,10 @@ And then activate it via the [`.hintrc`][hintrc] configuration file:
 [apache directory]: https://httpd.apache.org/docs/current/mod/core.html#directory
 [header directive]: https://httpd.apache.org/docs/current/mod/mod_headers.html#header
 [how to enable apache modules]: https://github.com/h5bp/server-configs-apache/tree/7eb30da6a06ec4fc24daf33c75b7bd86f9ad1f68#enable-apache-httpd-modules
-[htaccess is slow]: https://httpd.apache.org/docs/current/howto/htaccess.html#when
 [main apache conf file]: https://httpd.apache.org/docs/current/configuring.html#main
+[htaccess is slow]: https://httpd.apache.org/docs/current/howto/htaccess.html#when
```
Comment (molant) [inline packages/hint-x-content-type-options/README.md]: Can you please sort the links alphabetically?

File: packages/hint-x-content-type-options/README.md [hunk 2]
```diff
@@ -77,6 +76,69 @@ Content-Type: text/javascript; charset=utf-8
...
+  the logic in, for example, a [`<Directory>`][apache directory]
+  section in that file. This is usually the recommended way as
+  [using `.htaccess` files slows down][htaccess is slow] Apache!
+
+  If you don't have access to the main configuration file (quite
+  common with hosting services), add the snippets in a `.htaccess`
+  file in the root of the web site/app.
+
+For the complete set of configurations, not just for this rule, see
+the [Apache server configuration related documentation][apache config].
```
Comment (molant) [inline packages/hint-x-content-type-options/README.md]: I think you've missed to add the `apache config` link label at the end

**Comment by antross** [comment]
@molant given the README updates, this touched `packages/hint` so we're going to get a major version rev the next time we release due to the Breaking tag... Not sure if we already had other breaking changes queued up so this doesn't matter, or if you want to split the commit and re-write history, or just leave it.

**Comment by molant** [comment]
I missed that 😥
I'll rewrite the history.

**Comment by molant** [comment]
OK, should be fixed now. Nice catch @antross

**Pull request merged**